app/services/amr/validate_and_persist_readings_service.rb

```diff
@@ -0,0 +1,17 @@
+module Amr
+  class ValidateAndPersistReadingsService
```

Comment:
This used to do everything, it still does, but now delegates to the various services rather than doing it all by itself